### PR TITLE
Allow adding pod labels to gateway manifest

### LINKED ISSUE #43585: Add podLabels to Gateway Helm Chart
**Describe the feature request**
Add podLabels to Gateway Helm chart, for just adding labels to the pods (not the Deployment)

**Describe alternatives you've considered**
Kustomize, forking the chart

**Affected product area

[ ] Ambient
[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

**Affected features

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**
This issue was already opened back in September but got closed without any merge or denial comments.
https://github.com/istio/istio/issues/41057

We would also be happy to contribute in case of the acceptance of this feature.
( @blhagadorn )

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Fix https://github.com/istio/istio/issues/43585

### CODE REVIEW

**Review by howardjohn — CHANGES_REQUESTED** [review]
Why not just use .Values.labels which already exists

**Comment by hanxiaop** [comment]
> Why not just use .Values.labels which already exists

@howardjohn That param is to set labels for all resources, and here the `podLabels ` is only to set the template pod label of the deployment.

**Comment by howardjohn** [comment]
Is it so concerning to have the label in extra places? There are a bunch of different objects, I don't want `podLabels`, `serviceLabels`, `serviceAccountLabels`, etc

**Comment by hanxiaop** [comment]
> Is it so concerning to have the label in extra places? There are a bunch of different objects, I don't want `podLabels`, `serviceLabels`, `serviceAccountLabels`, etc

@howardjohn I see. It should be great to use the `.Values.labels` value.

File: manifests/charts/gateway/templates/deployment.yaml [hunk 1]
```diff
@@ -21,6 +21,9 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       labels:
+        {{- with .Values.labels }}
```
Comment (howardjohn) [inline manifests/charts/gateway/templates/deployment.yaml:24]: is this going to have duplicates with L31?

Comment (hanxiaop) [reply manifests/charts/gateway/templates/deployment.yaml:24]: Yes, it's included in .Values.labels. Thanks for the reminder. Fixed.

File: releasenotes/notes/gw-allow-labels.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+issue:
+  - 41057
+  - 43585
+releaseNotes:
+  - |
+    **Added** support for allowing labels to be added to the gateway pod template via `.Values.labels`.
```
Comment (dhawton) [inline releasenotes/notes/gw-allow-labels.yaml:9]:
```suggestion
    **Added** support for labels to be added to the Gateway pod template via `.Values.labels`.
```

File: manifests/charts/gateway/templates/deployment.yaml [hunk 3]
```diff
@@ -25,7 +25,7 @@ spec:
         {{- with .Values.revision }}
         istio.io/rev: {{ . }}
         {{- end }}
-        {{- include "gateway.selectorLabels" . | nindent 8 }}
+        {{- include "gateway.labels" . | nindent 8 }}
```
[the remark this answers is not in the document]
Comment (hanxiaop) [reply manifests/charts/gateway/templates/deployment.yaml]: Fixed.

**Review by howardjohn — APPROVED** [review]
thanks

**Pull request merged**